src/meshTools/AMIInterpolation/AMIInterpolation/AMIMethod/directAMI/directAMI.C

```diff
@@ -323,5 +362,76 @@ void Foam::directAMI<SourcePatch, TargetPatch>::calculate
     }
 }
 
+template<class SourcePatch, class TargetPatch>
+void Foam::directAMI<SourcePatch, TargetPatch>::calculateGpu
+(
+    labelListList& srcAddress,
+    scalarListList& srcWeights,
+    labelListList& tgtAddress,
+    scalarListList& tgtWeights
+)
+{
+    this->checkPatches();
+
+    // set initial sizes for weights and addressing
+    srcAddress.setSize(this->srcPatch_.size());
+    srcWeights.setSize(this->srcPatch_.size());
+    tgtAddress.setSize(this->tgtPatch_.size());
+    tgtWeights.setSize(this->tgtPatch_.size());
+
+    // check that patch sizes are valid
+    if (!this->srcPatch_.size())
+    {
+        return;
+    }
+    else if (!this->tgtPatch_.size())
+    {
+        WarningIn
+            (
+                "void Foam::AMIMethod<SourcePatch, TargetPatch>::initialise"
+                    "("
+                    "labelListList&, "
+                    "scalarListList&, "
+                    "labelListList&, "
+                    "scalarListList&, "
+                    "label&, "
+                    "label&"
+                    ")"
+            )
+        << this->srcPatch_.size() << " source faces but no target faces" << endl;
+
+        return;
+    }
+
+    const vectorgpuField& srcCf = this->srcPatch_.getFaceCentres();
 
+    // Is this correct? How do we get the list of points?
+    const pointgpuField& tgtPoints = this->tgtPatch_.getLocalPoints();
+    const pointgpuField& srcPoints = this->srcPatch_.getLocalPoints();
```

Comment:
Faces hold addresses to global point list so this should be the getPoints method.